evalml/automl/engine/engine_base.py

```diff
@@ -7,177 +7,205 @@
-            objectives_to_score = [automl.objective] + automl.additional_objectives
-            cv_pipeline = None
-            try:
-                logger.debug(f"\t\t\tFold {i}: starting training")
-                cv_pipeline = EngineBase.train_pipeline(pipeline, X_train, y_train, automl.optimize_thresholds, automl.objective)
-                logger.debug(f"\t\t\tFold {i}: finished training")
-                if automl.optimize_thresholds and pipeline.can_tune_threshold_with_objective(automl.objective) and automl.objective.can_optimize_threshold:
-                    logger.debug(f"\t\t\tFold {i}: Optimal threshold found ({cv_pipeline.threshold:.3f})")
-                logger.debug(f"\t\t\tFold {i}: Scoring trained pipeline")
-                scores = cv_pipeline.score(X_valid, y_valid, objectives=objectives_to_score)
-                logger.debug(f"\t\t\tFold {i}: {automl.objective.name} score: {scores[automl.objective.name]:.3f}")
-                score = scores[automl.objective.name]
-            except Exception as e:
-                if automl.error_callback is not None:
-                    automl.error_callback(exception=e, traceback=traceback.format_tb(sys.exc_info()[2]), automl=automl,
-                                          fold_num=i, pipeline=pipeline)
-                if isinstance(e, PipelineScoreError):
-                    nan_scores = {objective: np.nan for objective in e.exceptions}
-                    scores = {**nan_scores, **e.scored_successfully}
-                    scores = OrderedDict({o.name: scores[o.name] for o in [automl.objective] + automl.additional_objectives})
-                    score = scores[automl.objective.name]
-                else:
-                    score = np.nan
-                    scores = OrderedDict(zip([n.name for n in automl.additional_objectives], [np.nan] * len(automl.additional_objectives)))
-
-            ordered_scores = OrderedDict()
-            ordered_scores.update({automl.objective.name: score})
-            ordered_scores.update(scores)
-            ordered_scores.update({"# Training": y_train.shape[0]})
-            ordered_scores.update({"# Validation": y_valid.shape[0]})
-
-            evaluation_entry = {"all_objective_scores": ordered_scores, "score": score, 'binary_classification_threshold': None}
-            if is_binary(automl.problem_type) and cv_pipeline is not None and cv_pipeline.threshold is not None:
-                evaluation_entry['binary_classification_threshold'] = cv_pipeline.threshold
-            cv_data.append(evaluation_entry)
-        training_time = time.time() - start
-        cv_scores = pd.Series([fold['score'] for fold in cv_data])
-        cv_score_mean = cv_scores.mean()
-        logger.info(f"\tFinished cross validation - mean {automl.objective.name}: {cv_score_mean:.3f}")
-        return {'cv_data': cv_data, 'training_time': training_time, 'cv_scores': cv_scores, 'cv_score_mean': cv_score_mean}
```

Comment:
I'm assuming the impl here didn't change much?

Reply:
Correct!